source/utilities.cc

```diff
@@ -1724,11 +1724,103 @@ namespace aspect
+                                                    compute_table_indices(table_points, idx)));
+        }
+
+      // In case the data is specified on a grid that is equidistant
+      // in each coordinate direction, we only need to store
+      // (besides the data) the number of intervals in each direction and
+      // the begin- and endpoints of the coordinates.
+      // In case the grid is not equidistant, we need to keep
+      // all the coordinates in each direction, which is more costly.
+      std::array<unsigned int,dim> table_intervals;
+
+      coordinate_values_are_equidistant = true;
+      for (unsigned int d=0; d<dim; ++d)
+        {
+          table_intervals[d] = table_points[d]-1;
+
+          // The minimum and maximum coordinate values:
+          grid_extent[d].first = coordinate_values[d][0];
+          grid_extent[d].second = coordinate_values[d][table_points[d]-1];
+
+          const double grid_spacing = coordinate_values[d][1] - coordinate_values[d][0];
+
+          for (unsigned int n = 1; n < table_points[d]; n++)
+            {
+              const double current_grid_spacing = coordinate_values[d][n] - coordinate_values_[d][n-1];
+
+              AssertThrow(current_grid_spacing > 0.0,
+                          ExcMessage ("Coordinates in dimension "
+                                      + Utilities::int_to_string(d)
+                                      + " are not strictly ascending. "));
+
+              if (std::abs(current_grid_spacing - grid_spacing) > 0.005*(current_grid_spacing+grid_spacing))
+                coordinate_values_are_equidistant = false;
+            }
+        }
+
+      // For each data component, set up a GridData,
+      // its type depending on the read-in grid.
+      data.resize(components);
+      for (unsigned int c = 0; c < components; c++)
+        {
+          if (coordinate_values_are_equidistant)
+            data[c]
+              = std_cxx14::make_unique<Functions::InterpolatedUniformGridData<dim>> (grid_extent,
+                                                                                     table_intervals,
+                                                                                     raw_data[c]);
+          else
+            data[c]
+              = std_cxx14::make_unique<Functions::InterpolatedTensorProductGridData<dim>> (coordinate_values,
+                                                                                           raw_data[c]);
+        }
+    }
+
+
+
     template <int dim>
     void
     AsciiDataLookup<dim>::load_file(const std::string &filename,
                                     const MPI_Comm &comm)
     {
+      // grab the values already stored in this class (if they exist), this way we can
+      // check if somebody changes the size of the table over time and error out (see below)
+      TableIndices<dim> table_points = this->table_points;
```

Comment:
Uh, that's going to confuse someone at some point. Just call the new variable `new_table_points` or something like that.

Reply:
That is a good point. Fixed.